src/Pos/Block/Arbitrary.hs

```diff
@@ -220,8 +219,7 @@ newtype BlockHeaderList ssc = BHL
     } deriving (Eq)
 
 instance T.BiSsc ssc => Show (BlockHeaderList ssc) where
-    show =
-        concat . intersperse "\n" . map (formatToString build) . uncurry zip . getHeaderList
+    show = intercalate "\n" . map (formatToString build) . uncurry zip . getHeaderList
```

Comment:
@rockbmb have you applied hlint again after applying it for the first time? It doesn't apply hints recursively.